src/Microsoft.FeatureManagement/ConfigurationFeatureDefinitionProvider.cs

```diff
@@ -199,17 +205,19 @@ We support
 
         private IEnumerable<IConfigurationSection> GetFeatureDefinitionSections()
         {
-            const string FeatureManagementSectionName = "FeatureManagement";
+            //
+            // Look for feature definitions under the "FeatureManagement" section
+            IConfigurationSection featureManagementConfigurationSection = _configuration.GetSection(FeatureManagementSectionName);
 
-            if (_configuration.GetChildren().Any(s => s.Key.Equals(FeatureManagementSectionName, StringComparison.OrdinalIgnoreCase)))
+            if (featureManagementConfigurationSection.Exists())
             {
-                //
-                // Look for feature definitions under the "FeatureManagement" section
-                return _configuration.GetSection(FeatureManagementSectionName).GetChildren();
+                return featureManagementConfigurationSection.GetChildren();
             }
             else
             {
-                return _configuration.GetChildren();
+                _logger.LogWarning($"No configuration section named '{FeatureManagementSectionName}' was found.");
```

Comment:
There's just no features defined. That's fine. We have a separate flag to throw or not if a requested feature is not found.

Reply:
For developers, this code gets hit on each `IsEnabledAsync` call and on direct calls to `GetFeatureDefinitionAsync`.

For `IsEnabledAsync`, we do log that the feature wasn't found. I'd be okay to take it out, but I think there is some value in indicating that it never found a FeatureManagement section at all.

But for direct calls to `GetFeatureDefinitionAsync`, this is the only warning. I suppose the resulting object being empty is indication that something is wrong.

What if we kept it but adjusted it to `LogInformation`?

Reply:
LogDebug or LogTrace perhaps.

Reply:
Adjusted to LogDebug